controllers/apps/operations/datascript.go

```diff
@@ -0,0 +1,283 @@
+	return realAffectedComponentMap(opsRequest.Spec.GetDataScriptComponentNameSet())
+}
+
+// getScriptContent will get script content from script or scriptFrom
+func getScriptContent(reqCtx intctrlutil.RequestCtx, cli client.Client, spec *appsv1alpha1.ScriptSpec) ([]string, error) {
+	script := make([]string, 0)
+	if len(spec.Script) > 0 {
+		script = append(script, spec.Script...)
+	}
+	if spec.ScriptFrom == nil {
+		return script, nil
+	}
+	configMapsRefs := spec.ScriptFrom.ConfigMapRef
+	secretRefs := spec.ScriptFrom.SecretRef
+
+	if len(configMapsRefs) > 0 {
+		obj := &corev1.ConfigMap{}
+		for _, cm := range configMapsRefs {
+			if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: reqCtx.Req.Namespace, Name: cm.Name}, obj); err != nil {
+				return nil, err
+			}
+			script = append(script, obj.Data[cm.Key])
+		}
+	}
+
+	if len(secretRefs) > 0 {
+		obj := &corev1.Secret{}
+		for _, secret := range secretRefs {
+			if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: reqCtx.Req.Namespace, Name: secret.Name}, obj); err != nil {
+				return nil, err
+			}
+			if obj.Data[secret.Key] == nil {
+				return nil, fmt.Errorf("secret %s/%s does not have key %s", reqCtx.Req.Namespace, secret.Name, secret.Key)
+			}
+			secretData := string(obj.Data[secret.Key])
+			// trim the last \n
+			if len(secretData) > 0 && secretData[len(secretData)-1] == '\n' {
+				secretData = secretData[:len(secretData)-1]
+			}
+			script = append(script, secretData)
+		}
+	}
+	return script, nil
+}
+
+func getTargetPod(reqCtx intctrlutil.RequestCtx, cli client.Client, clusterObjectKey client.ObjectKey, componentName string) (string, error) {
+	// get svc
+	service := &corev1.Service{}
+	serviceName := fmt.Sprintf("%s-%s", clusterObjectKey.Name, componentName)
+	if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: clusterObjectKey.Namespace, Name: serviceName}, service); err != nil {
+		return "", err
+	}
+	// get selector from service
+	selector := service.Spec.Selector
+	// get pods by selector
+	podList := &corev1.PodList{}
+	if err := cli.List(reqCtx.Ctx, podList, client.InNamespace(clusterObjectKey.Namespace), client.MatchingLabels(selector)); err != nil {
+		return "", err
+	}
+
+	if len(podList.Items) == 0 {
+		return "", fmt.Errorf("no pod found by service %s", componentName)
+	}
+
+	// get the first pod
+	pod := podList.Items[0]
+	podIP := pod.Status.PodIP
+	if len(podIP) == 0 {
+		return "", fmt.Errorf("pod %s has no podIP", pod.Name)
+	}
+	return podIP, nil
+}
+
+func createDataScriptJob(cluster *appsv1alpha1.Cluster, component *appsv1alpha1.ClusterComponentSpec, ops *appsv1alpha1.OpsRequest, scripts string) (*batchv1.Job, error) {
+	jobName := fmt.Sprintf("%s-%s-%s", "script", ops.Name, component.Name)
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: cluster.Namespace,
+		},
+	}
```

Comment:
Verify that the character length of jobName, cannot exceed 63

Reply:
refined job name